pkg/cmd/repo/archive/archive_test.go

```diff
@@ -66,80 +51,85 @@ func TestNewCmdArchive(t *testing.T) {
+							{ "id": "THE-ID",
+					   		"isArchived": %s} 
+						} }`
 	tests := []struct {
 		name       string
 		opts       ArchiveOptions
 		httpStubs  func(*httpmock.Registry)
+		askStubs   func(*prompt.AskStubber)
 		isTTY      bool
 		wantStdout string
 		wantStderr string
 	}{
 		{
 			name:       "unarchived repo tty",
-			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
 			wantStdout: "✓ Archived repository OWNER/REPO\n",
-			isTTY:      true,
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
+			isTTY: true,
+			opts:  ArchiveOptions{RepoArg: "OWNER/REPO"},
 			httpStubs: func(reg *httpmock.Registry) {
 				reg.Register(
 					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": false} } }`))
+					httpmock.StringResponse(fmt.Sprintf(queryResponse, "false")))
 				reg.Register(
 					httpmock.GraphQL(`mutation ArchiveRepository\b`),
 					httpmock.StringResponse(`{}`))
 			},
 		},
 		{
-			name:  "unarchived repo notty",
-			opts:  ArchiveOptions{RepoArg: "OWNER/REPO"},
-			isTTY: false,
+			name:       "infer base repo",
+			wantStdout: "✓ Archived repository OWNER/REPO\n",
+			opts:       ArchiveOptions{},
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
+			isTTY: true,
 			httpStubs: func(reg *httpmock.Registry) {
 				reg.Register(
 					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": false} } }`))
+					httpmock.StringResponse(fmt.Sprintf(queryResponse, "false")))
 				reg.Register(
 					httpmock.GraphQL(`mutation ArchiveRepository\b`),
 					httpmock.StringResponse(`{}`))
 			},
 		},
 		{
 			name:       "archived repo tty",
-			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
 			wantStderr: "! Repository OWNER/REPO is already archived\n",
-			isTTY:      true,
-			httpStubs: func(reg *httpmock.Registry) {
-				reg.Register(
-					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": true } } }`))
-			},
-		},
-		{
-			name:       "archived repo notty",
 			opts:       ArchiveOptions{RepoArg: "OWNER/REPO"},
-			isTTY:      false,
-			wantStderr: "! Repository OWNER/REPO is already archived\n",
+			askStubs: func(q *prompt.AskStubber) {
+				q.StubOne(true)
+			},
 			httpStubs: func(reg *httpmock.Registry) {
 				reg.Register(
 					httpmock.GraphQL(`query RepositoryInfo\b`),
-					httpmock.StringResponse(`{ "data": { "repository": {
-												"id": "THE-ID",
-												"isArchived": true } } }`))
+					httpmock.StringResponse(fmt.Sprintf(queryResponse, "true")))
 			},
 		},
 	}
 
 	for _, tt := range tests {
+		repo, _ := ghrepo.FromFullName("OWNER/REPO")
+		tt.opts.BaseRepo = func() (ghrepo.Interface, error) {
+			return repo, nil
+		}
```

Comment:
Just as a purely style nit, I like to define all the `tt.opts` next to each other so it is easy to see all of them at a glance.